### ISSUE TITLE
the commitment: one more residual

### ISSUE THREAD

**Issue by liangbingzhao** [lucidrains/video-diffusion-pytorch#14]
After updating this commitment, the color of the sampled video fades(not know why); i am using the ucf101 dataset, unconditional training with a 10k step warmup.

**Comment by dhruv-nathawani** [comment]
Were you able to get UCF101 unconditional training to work?
I am getting moving colored patches for the sampled video

**Comment by liangbingzhao** [comment]
> Were you able to get UCF101 unconditional training to work? I am getting moving colored patches for the sampled video

yeah, just adjust unet and lr should work. I just use the ApplyEyeMakeup for training. If you use the whole dataset, maybe train longer? I haven't tested on the whole yet.

**Comment by dhruv-nathawani** [comment]
I am trying whole dataset and it gives random colored artifacts

**Comment by liangbingzhao** [comment]
is your loss converged? and maybe train longer? I train with eyemakeup folder, and need 60-70k to get good 64*64 results, need 150+k to get good 128*128 results. And the longer the better. so maybe you should let it train longer? I guess?

**Comment by dhruv-nathawani** [comment]
Okay makes sense!
I trained, batch_size = 64 on 8 A100 GPU for 20k iterations on eyemakeup folder (64x64) and got some results but not great. 
LR = 1e-03, also implemented skip 1 frame like the original paper.

It takes more than 2-3 days to get to 20k iterations, this is super super slow !!!

Do you mind sharing what U-Net modifications, LR and batch_size you have used?

**Comment by liangbingzhao** [comment]
lr = 1e-4, 
batch_size=8(2 1080ti), 
10k cosine warmup, 
model = Unet3D(dim = 64,dim_mults = (1, 2, 4, 8, 8))
prob_focus_present I don't find matter a lot, num it 0.0 or 0.2 produce similar quality results
you give it a try?

**Comment by liangbingzhao** [comment]
well, it shouldn't be that slow. I can train 70k using less than 2 days.(with DP accelerate)

**Comment by dhruv-nathawani** [comment]
Thank you, I can try these settings on 16 A100 GPUs for :
1. eyemakeup folder
2. entire ucf101 dataset

I also set prob_focus_present = 0.2

**Comment by dhruv-nathawani** [comment]
> well, it shouldn't be that slow. I can train 70k using less than 2 days.(with DP accelerate)

Oh really? I just use nn.DataParallel and run the model

Hmm wondering if I can try DP accelerate as well then

**Comment by dhruv-nathawani** [comment]
Also your learning rate seems high compared to mine. Interesting
1e-04 for batch_size = 8

Original paper is 3e-04 for batch_size = 128

**Comment by dhruv-nathawani** [comment]
> well, it shouldn't be that slow. I can train 70k using less than 2 days.(with DP accelerate)

Did you use accumulate_gradient=2?

**Comment by liangbingzhao** [comment]
aha, i am sorry, i mean using dataparallel to accelerate. sorry make you confused

**Comment by dhruv-nathawani** [comment]
Also I guess batch_size = 8, 70k iterations is approx equivalent to batch_size = 64, 10k iterations

**Comment by liangbingzhao** [comment]
> Also your learning rate seems high compared to mine. Interesting 1e-04 for batch_size = 8
> 
> Original paper is 3e-04 for batch_size = 128

1e-4 is smaller than your 1e-3. and the paper mention their training settings?? maybe I should check it out.
additionally, if you make your batch size smaller, shouldn't you make your lr smaller, not bigger?

**Comment by dhruv-nathawani** [comment]
Yes, LR should be smaller but I was thinking more like linear scaling down (most blogs recommend)

https://stackoverflow.com/questions/53033556/how-should-the-learning-rate-change-as-the-batch-size-change

**Comment by liangbingzhao** [comment]
yeah, i'm just first 10k cosine warmup, then also linear down from 1e-4.
I tried some bigger lr(5e-4) when producing 128\*128, finding that really hard to converge, and results are bad.
Hope these can help you.

**Comment by dhruv-nathawani** [comment]
Yes definitely, thank you!

Also did you try "l1" vs "l2" loss?
I noticed "l2" gives better result (also used in original paper)

**Comment by liangbingzhao** [comment]
I use l1, not have time to try l2, I will give it a try!

**Comment by dhruv-nathawani** [comment]
Thanks, I will share some samples once I get good results!

**Comment by liangbingzhao** [comment]
looking forward to your results~

**Comment by liangbingzhao** [comment]
btw, where do you find their training setting? I check the paper and the github.io, not finding them 😢

**Comment by dhruv-nathawani** [comment]
Its on page 14 of the supplemental section ("Details and Hyper-parameters") of the paper

**Comment by liangbingzhao** [comment]
oh, damn. I read and preserve their first version paper, not knowing their update. thx!!

**Comment by liangbingzhao** [comment]
@dhruv-nathawani  have you achieve any good results? I selected 10 categories of ucf101 to train a 64\*64 model, batch_size = 8. After training 600K steps, got few good results.

**Comment by oxjohanndiep** [comment]
@martinriven can you show us some of your results?

**Comment by dhruv-nathawani** [comment]
@martinriven I got good results for ApplyEyeMakeup, but when I tried training with 3 different video actions from UCF101 it did not work, would you mind sharing your architecture (parameters we discussed earlier) and results?

Here are some results for ApplyEyeMakeup (they seem to have overfit, which is expected training with just 145 videos) : 

![114](https://user-images.githubusercontent.com/36652102/187254986-e52f4830-4a9e-4281-9145-d0860ea63c61.gif)  ![164](https://user-images.githubusercontent.com/36652102/187254845-28b70623-a3b5-414a-a8ff-3f49e497d3c0.gif)            ![168](https://user-images.githubusercontent.com/36652102/187254847-8789f132-220d-48fd-9ccc-8c3814f5d74f.gif)

**Comment by martinriven** [comment]
@dhruv-nathawani  sry so late to reply. Well my computational resource is limited, I could only train on 10 frames, that may contribute to part of the reason? My ApplyEyeMakeup results are like this:

https://user-images.githubusercontent.com/55656865/187843295-3eb971db-abd0-41fa-8c5b-c6a03e3c2ac0.mp4


https://user-images.githubusercontent.com/55656865/187843431-ddf7cb25-e9f9-4856-85e9-61f487aa3b56.mp4



https://user-images.githubusercontent.com/55656865/187843760-be2c39b2-84f6-4b96-81c3-dbc038ea23c8.mp4

The architecture, I barely change it. About the parameters, after the discussion, I tried l2 loss, and it really helped. About other video actions, I haven't tried to train only one other action, I just tried to train 10 actions together, and got sth like this(which looks really bad).




https://user-images.githubusercontent.com/55656865/187844521-f71d8c6d-a2fd-4259-aead-902269b3db54.mp4


https://user-images.githubusercontent.com/55656865/187844971-24ee5eb3-bec7-4a22-8244-9e95a91295f4.mp4

**Comment by martinriven** [comment]
why my mp4 cannot show on here? 😢
